int64 x_sector
int64 y_sector
int64 x
int64 y
